lib/src/visible_range.dart

```diff
@@ -57,29 +91,57 @@ abstract class VisibleRange {
 
     return velocity.abs() > tolerance.velocity ? 0.5 * velocity.sign : 0.0;
   }
+  
+  bool dateInsideAvailableRange(LocalDate date);
+  num getDaysBefore(LocalDate date);
+  num getDaysAfter(LocalDate date);
```

Comment:
These need to provide default implementations to not break custom subclasses. It should suffice to just move the implementation in `WeekVisibleRange` up here

Reply:
added default implementation for all of them